### PR TITLE
Use golang.org/x/time for rate limiting

### PR DESCRIPTION
**Pull request by andrewkroh**

### CODE REVIEW

File: auditbeat/module/file_integrity/scanner.go [hunk 1]
```diff
@@ -155,13 +154,24 @@ func (s *scanner) throttle(fileSize uint64) {
...
+	delay := reservation.Delay()
+	if delay == 0 {
+		return
+	}
+
+	timer := time.NewTimer(delay)
+	defer timer.Stop()
+	select {
+	case <-s.done:
+	case <-timer.C:
```
Comment (ph) [inline auditbeat/module/file_integrity/scanner.go]: Didn't know about that library, TIL.

File: auditbeat/module/file_integrity/scanner.go [hunk 2]
```diff
@@ -55,10 +54,10 @@ func (s *scanner) Start(done <-chan struct{}) (<-chan Event, error) {
 				s.config.ScanRatePerSec,
 				s.config.MaxFileSize)
 
-		s.tokenBucket = ratelimit.NewBucketWithRate(
-			float64(s.config.ScanRateBytesPerSec)/2., // Fill Rate
-			int64(s.config.MaxFileSizeBytes))         // Max Capacity
-		s.tokenBucket.TakeAvailable(math.MaxInt64)
+		s.tokenBucket = rate.NewLimiter(
```
Comment (ph) [inline auditbeat/module/file_integrity/scanner.go]: Not sure we were using half the `ScanRateBytesPerSec` before? Everything else seems OK for the migration.

Comment (andrewkroh) [reply auditbeat/module/file_integrity/scanner.go]: That was a fudge factor. 🤦‍♂️

**Comment by ph** [comment]
LGTM failures is fixed in a newer release of docker-py

**Comment by ph** [comment]
We will need to backport to 6.2

**Pull request merged**